### PR TITLE
Split `compute` package into few smaller ones

### PR DESCRIPTION
**Pull request by nfx**
As the result of this refactoring, more purpose-driven packages are created:
* `clusters` for ClustersAPI (+ node types, spark versions)
* `commands` for CommandsAPI (used for mounts and sql permissions)
* `libraries` for `databricks_cluster` and `databricks_job`
* `pools` for `databricks_instance_pool`
* `policies` for `databricks_cluster_policy`
* `pipelines` for `databricks_pipeline`

This refactoring also moves all TestAcc into `acceptance` sub-packages as much as possible, in order to reason about dependency tree easier.

This pull request also introduces alphabetical order for resources in provider.go

### CODE REVIEW

**Comment by nfx** [comment]
libraries test coverage would be handled in #904

File: clusters/clusters_api.go [hunk 1]
```diff
@@ -298,6 +359,47 @@ type Cluster struct {
...
+	master := cluster.SparkConf["spark.master"]
+	resourceClass := cluster.CustomTags["ResourceClass"]
+	if profile == "singleNode" && strings.HasPrefix(master, "local") && resourceClass == "SingleNode" {
+		return nil
+	}
+	return fmt.Errorf("NumWorkers could be 0 only for SingleNode clusters. See https://docs.databricks.com/clusters/single-node.html for more details")
+}
+
+// ModifyRequest helps remove all request fields that should not be submitted when instance pool is selected.
+func (cluster *Cluster) ModifyRequest() {
```
Comment (alexott) [inline clusters/clusters_api.go:377]: Maybe give it better name? like `modifyClusterOnInstancePool`, or something like

File: clusters/clusters_api_test.go [hunk 2]
```diff
@@ -569,57 +571,6 @@ func TestPermanentDelete_Pinned(t *testing.T) {
 	require.NoError(t, err)
 }
 
-func TestAccListClustersIntegration(t *testing.T) {
```
Comment (alexott) [inline clusters/clusters_api_test.go:572]: why test is removed?

File: commands/commands_test.go [hunk 3]
```diff
@@ -513,37 +513,37 @@ func TestCommandsAPIExecute_NoCommandResults(t *testing.T) {
 	})
 }
 
-func TestAccContext(t *testing.T) {
```
Comment (alexott) [inline commands/commands_test.go:516]: and this one?

File: storage/mounts.go [hunk 4]
```diff
@@ -94,8 +94,8 @@ func (mp MountPoint) Mount(mo Mount, client *common.DatabricksClient) (source st
 				raise e
 		mount_source = safe_mount("/mnt/%s", "%v", %s, "%s")
 		dbutils.notebook.exit(mount_source)
-	`, mp.name, mo.Source(), extraConfigs, mp.encryptionType) // lgtm[go/unsafe-quoting]
-	result := mp.exec.Execute(mp.clusterID, "python", command)
+	`, mp.Name, mo.Source(), extraConfigs, mp.EncryptionType) // lgtm[go/unsafe-quoting]
```
Comment (alexott) [inline storage/mounts.go:97]: it's interesting that this directive doesn't help

**Review by alexott — APPROVED** [review]
lgtm, just small comments

File: clusters/clusters_api_test.go (hunk 2 above)
Comment (nfx) [reply clusters/clusters_api_test.go:572]: It's moved

File: storage/mounts.go (hunk 4 above)
Comment (nfx) [reply storage/mounts.go:97]: where did you see warning triggered? bot didn't comment and security tab is clear 🤷‍♂️

Comment (alexott) [reply storage/mounts.go:97]: https://github.com/databrickslabs/terraform-provider-databricks/pull/914/checks?check_run_id=4199083573

File: clusters/clusters_api_test.go (hunk 2 above)
Comment (nfx) [reply clusters/clusters_api_test.go:572]: moved to `clusters/acceptance/clusters_api_test.go`

File: commands/commands_test.go (hunk 3 above)
Comment (nfx) [reply commands/commands_test.go:516]: moved to `commands/acceptance/commands_test.go`

**Pull request merged**